app/test/shared/search_service_test.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:test/test.dart';
+
+import 'package:pub_dartlang_org/shared/search_service.dart';
+
+void main() {
+  group('SearchOrder enum', () {
+    test('serialization', () {
+      for (var value in SearchOrder.values) {
+        final String serialized = serializeSearchOrder(value);
+        expect(serialized, isNotEmpty);
+        final SearchOrder deserialized = parseSearchOrder(serialized);
+        expect(deserialized, value);
+      }
```

Comment:
If you write a test, then you might as well add a case for `parseSearchOrder('foobar', defaultsTo: ...)` :)

Reply:
added